src/main/java/seedu/address/model/country/Country.java

```diff
@@ -1,16 +1,69 @@
 package seedu.address.model.country;
 
-public enum Country {
-//    TODO: Add implementation
-//    private Set<Note> noteList;
-//    public Set<Note> getNoteList() {
-//        if (noteList != null) {
-//            return Collections.unmodifiableSet(noteList);
-//        } else {
-//            return new HashSet<>();
-//        }
-//    }
-//    public void setNoteList(Set<Note> noteList) {
-//        this.noteList = noteList;
-//    }
+import static java.util.Objects.requireNonNull;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Locale;
+
+import seedu.address.model.note.Note;
+
+/**
+ * A representation of a Country that can be identified by a 2-letter ISO3166 country-code or by its country
+ * name. It contains a list of country notes.
+ */
+public class Country {
+
+    private final String countryName;
+    private final String countryCode;
+    private final ArrayList<Note> countryNotes;
```

Comment:
```suggestion
    private final String COUNTRY_NAME;
    private final String COUNTRY_CODE;
    private final ArrayList<Note> COUNTRY_NOTES;
```

From the [convention guide](https://se-education.org/guides/conventions/java/intermediate.html): Constant names must be all uppercase using underscore to separate words. I'm not sure if countryNotes should be marked `final` actually, since we are going to modify it in the `addCountryNote` function.

Also, I feel that the notes should be stored in something like a LinkedHashSet, since we don't really want to add in duplicate notes, and its easier to check if its a HashSet.

Reply:
actually all those variables aren't constants in the sense that for each instance of `Country` the `countryName`, etc can be different. Not sure if they are still considered constants?

Agree with using HashSet to avoid duplicate Note.

Reply:
Hmmmm seems like by `constant` they mean static final only. Static only or final only aren't counted as constants, even though one of the examples in the website has an example of final only variables being set to all caps + underscores.

No need to change bah